### PR TITLE
aws auth: add missing auth methods

### PR DESCRIPTION
**Pull request by famarting**
# Description

Testing AWS IAM auth with postgres and missing the assume role feature that its documented as part of the aws auth profile

It all started because this error check here is bad https://github.com/dapr/components-contrib/blob/main/common/authentication/aws/aws.go#L133
if there is no error, `errors.Is(err, config.SharedConfigAssumeRoleError{}.Err)` always evaluates to true... because when you do `config.SharedConfigAssumeRoleError{}.Err` then `Err` is nil!

I started fixing that but then looking at the kafka pubsub component https://github.com/dapr/components-contrib/blob/main/common/component/kafka/auth.go#L99 , kafka supports more fields, so it made sense to also support these for postgres.

docs updated here https://github.com/dapr/docs/pull/4429

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: common/authentication/postgresql/metadata.go [hunk 1]
```diff
@@ -86,16 +86,30 @@ func (m *PostgresAuthMetadata) InitWithMetadata(meta map[string]string, opts Ini
...
-	// in the event users are leveraging the credential files for an access token.
+	// only region is mandatory
+
 	awsAccessKey, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSAccessKey")
 	awsSecretKey, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSSecretKey")
-	return awsRegion, awsAccessKey, awsSecretKey, nil
+	awsSessionToken, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSSessionToken")
+
+	awsIamRoleArn, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "awsIamRoleArn")
+	awsStsSessionName, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "awsStsSessionName")
```
Comment (sicoyle) [inline common/authentication/postgresql/metadata.go:101]:
```suggestion
	sessionName, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "sessionName")
```

I don't think we need to expose to end users the underlying mechanisms leveraging Sts through which one can assume a new role with new creds

File: common/authentication/aws/aws.go [hunk 2]
```diff
@@ -118,40 +119,66 @@ type AWSIAMAuthOptions struct {
 	Region           string          `json:"region" mapstructure:"region"`
 	AccessKey        string          `json:"accessKey" mapstructure:"accessKey"`
 	SecretKey        string          `json:"secretKey" mapstructure:"secretKey"`
+	SessionToken     string          `json:"sessionToken" mapstructure:"sessionToken"`
+
+	AssumeRoleArn         string `json:"assumeRoleArn" mapstructure:"assumeRoleArn"`
```
Comment (famarting) [inline common/authentication/aws/aws.go:124]: agreed to change this fields tags, but they are not used for anything

File: common/authentication/postgresql/metadata.go [hunk 3]
```diff
@@ -86,16 +86,30 @@ func (m *PostgresAuthMetadata) InitWithMetadata(meta map[string]string, opts Ini
...
-	// Note: access key and secret keys can be optional
-	// in the event users are leveraging the credential files for an access token.
+	// only region is mandatory
+
 	awsAccessKey, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSAccessKey")
 	awsSecretKey, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSSecretKey")
-	return awsRegion, awsAccessKey, awsSecretKey, nil
+	awsSessionToken, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSSessionToken")
+
+	awsIamRoleArn, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "awsIamRoleArn")
```
[the remark this answers is not in the document]
Comment (famarting) [reply common/authentication/postgresql/metadata.go:100]: this cannot be changed, it must match with metadata.yaml , metadata.yaml has been set to match the existing metadata names for the kafka component, so we have consistency

File: common/authentication/postgresql/metadata.go (hunk 1 above)
Comment (famarting) [reply common/authentication/postgresql/metadata.go:101]: this cannot be changed, it must match with metadata.yaml , metadata.yaml has been set to match the existing metadata names for the kafka component, so we have consistency

**Comment by sicoyle** [comment]
This PR adds "missing auth methods" that are not actually missing. These are for a different auth profile, that I've added for us to in this PR: https://github.com/dapr/components-contrib/pull/3606 when moving the Kafka component to use the built in AWS auth profiles. 

Please do not merge this PR until we get an official auth profile for assuming an IAM role. We should not continue down the path of adding new disjoint fields to kafka/postgres specific to IAM until they get moved to the common AWS auth profiles. Otherwise, we would have to keep deprecating old component specific IAM fields, adding to maintainer overhead.

**Comment by famarting** [comment]
ok, closing

added feedback to the kafka and postgres PR. I'm missing the assume role functionality for the postgres components

**Pull request closed without merging**